### ISSUE TITLE
Flaky test: ResponseTests.ResponseFormats_Nginx_ContentLength

### ISSUE THREAD

**Issue by ryanbrandenburg** [aspnet/ServerTests#107]
This test [fails](http://aspnetci/viewLog.html?buildId=435507&buildTypeId=Releases_21Public_UbuntuUniverse) sometimes:

```
System.Exception : Failed to start nginx
   at Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer.SetupNginx(String redirectUri, Uri originalUri)
   at Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer.<DeployAsync>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at ServerComparison.FunctionalTests.ResponseTests.<ResponseFormats>d__16.MoveNext() in /_/test/ServerComparison.FunctionalTests/ResponseTests.cs:line 175
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
------- Stdout: -------
| [2018-03-29T15:03:55] TestLifetime Information: Starting test ResponseFormats_Nginx_ContentLength_Nginx_CoreClr_x64_Portable
| [2018-03-29T15:03:55] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: Executing dotnet run --no-restore --framework netcoreapp2.0 -- --server.urls http://localhost:42703/  --server Microsoft.AspNetCore.Server.Kestrel
| [2018-03-29T15:03:55] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: SET ASPNETCORE_ENVIRONMENT=Responses
| [2018-03-29T15:03:55] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: SET ASPNETCORE_DETAILEDERRORS=true
| [2018-03-29T15:03:55] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: Started dotnet. Process Id : 68074
| [2018-03-29T15:03:55] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: host process ID 67989 shut down
| [2018-03-29T15:03:55] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: Host process shutting down.
| [2018-03-29T15:03:57] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: dotnet stdout: Using launch settings from /mnt/work/33bdfc1cae7b2a38/modules/ServerTests/test/ServerComparison.TestSites/Properties/launchSettings.json...
| [2018-03-29T15:03:57] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: dotnet stderr: The launch profile "(Default)" could not be applied.
| [2018-03-29T15:03:58] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: dotnet stderr: A usable launch profile could not be located.
| [2018-03-29T15:03:58] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: dotnet stdout: Hosting environment: Responses
| [2018-03-29T15:03:58] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: dotnet stdout: Content root path: /mnt/work/33bdfc1cae7b2a38/modules/ServerTests/test/ServerComparison.TestSites/bin/Release/netcoreapp2.0/
| [2018-03-29T15:03:58] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: dotnet stdout: Now listening on: http://localhost:42703
| [2018-03-29T15:03:58] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: dotnet stdout: Application started. Press Ctrl+C to shut down.
| [2018-03-29T15:03:58] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx start stderr: nginx: [alert] could not open error log file: open() "/var/log/nginx/error.log" failed (13: Permission denied)
| [2018-03-29T15:03:58] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx start stderr: nginx: [emerg] bind() to 0.0.0.0:33012 failed (98: Address already in use)
| [2018-03-29T15:03:59] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx start stderr: nginx: [emerg] bind() to 0.0.0.0:33012 failed (98: Address already in use)
| [2018-03-29T15:03:59] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx start stderr: nginx: [emerg] bind() to 0.0.0.0:33012 failed (98: Address already in use)
| [2018-03-29T15:04:00] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx start stderr: nginx: [emerg] bind() to 0.0.0.0:33012 failed (98: Address already in use)
| [2018-03-29T15:04:00] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx start stderr: nginx: [emerg] bind() to 0.0.0.0:33012 failed (98: Address already in use)
| [2018-03-29T15:04:01] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx start stderr: nginx: [emerg] still could not bind()
| [2018-03-29T15:04:01] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx stop stderr: nginx: [alert] could not open error log file: open() "/var/log/nginx/error.log" failed (13: Permission denied)
| [2018-03-29T15:04:01] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx stop stderr: 2018/03/29 15:04:01 [notice] 68145#68145: signal process started
| [2018-03-29T15:04:01] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: nginx stop stderr: 2018/03/29 15:04:01 [error] 68145#68145: open() "/mnt/work/33bdfc1cae7b2a38/modules/ServerTests/test/ServerComparison.TestSites/57356049-afb1-4a75-908f-fbd30f43df02.nginx.pid" failed (2: No such file or directory)
| [2018-03-29T15:04:01] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: nginx stop command issued
| [2018-03-29T15:04:01] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: Attempting to cancel process 68074
| [2018-03-29T15:04:01] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: dotnet stdout: Application is shutting down...
| [2018-03-29T15:04:01] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Warning: Unable to terminate the host process with process Id '68074
| [2018-03-29T15:04:01] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer Information: [Time]: Total time taken for this test variation '0' seconds
| [2018-03-29T15:04:01] TestLifetime Information: Finished test ResponseFormats_Nginx_ContentLength_Nginx_CoreClr_x64_Portable in 5.3439744s
```

Future instances of this failure will appear [here](http://aspnetci/httpAuth/app/rest/builds?locator=tag:NGinxStartFailed,sinceDate:201710010T000000-0001&fields=build(startDate,buildTypeId,status,branchName,webUrl,tags)).

**Comment by Tratcher** [comment]
@mikeharder also related to port in use errors

**Comment by ryanbrandenburg** [comment]
HelloWorld_Nginx [failed again](http://aspnetci/viewLog.html?buildId=514027&buildTypeId=Releases_21Public_UbuntuUniverse).

**Comment by ryanbrandenburg** [comment]
ResponseCompression_NoCompression [failed again](http://aspnetci/viewLog.html?buildId=514893&buildTypeId=XPlat_Linux_UbuntuUniverseCoreCLR).

**Comment by ryanbrandenburg** [comment]
ResponseFormats_Chunked [failed again](http://aspnetci/viewLog.html?buildId=516083&buildTypeId=Releases_22xPublic_UbuntuUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
How was this resolved?

**Comment by Tratcher** [comment]
Duplicate of https://github.com/aspnet/ServerTests/issues/135, resolved by aspnet/Hosting#1515
